### ISSUE TITLE
Add support for Render Props to Tab

### ISSUE THREAD

**Issue by eljefe223** [Microsoft/fast-dna#1166]
**What is your feature request? Please describe.**
Currently Tabs relies on slots to render. Slots are prone to falling out of sync with component updates because they are loosely types. Moving to render props will make the the tab component more intuitive.

**Describe the solution you'd like**
Add render props to tabs component to make tabs API more intuitive and easier to implement.

**Additional context**
More discussion on slots vs render props can be read [here](https://github.com/Microsoft/fast-dna/issues/1135)

**Comment by nicholasrice** [comment]
In the "new way" the TabItems still have slot attributes - is that intentional?

**Comment by eljefe223** [comment]
@nicholasrice no! Thanks for catching that

**Comment by nicholasrice** [comment]
yep - this looks good to me. we should still preserve the slot functionality if possible - if it's not we'll need to make a breaking change.
